### ISSUE TITLE
Default to including the new homebrew path for apple silicon (m1) macs

### ISSUE THREAD

**Issue by jeffwidman** [sorin-ionescu/prezto#1988]
On an Apple Silicon / M1 mac, Homebrew now installs binaries to `/opt/homebrew/{bin,sbin}` instead of `/usr/local/{bin,sbin}`.

Given the popularity of homebrew, IMO we should update Prezto to default to including this new path.

This could be done at several different abstraction layers:
* the `homebrew` module
* hardcoding the path (as demonstrated here: https://github.com/sorin-ionescu/prezto/pull/1974/files)

The downside of adding it to the homebrew module is that it's only loaded during an interactive session: https://github.com/sorin-ionescu/prezto/blob/20641f3447e9caef169c8210ff540921419ce448/runcoms/zshrc#L8-L11

Whereas I have a few places in my runcoms where I set env vars using something like:

```shell
export VIRTUALENVWRAPPER_PYTHON=$(brew --prefix)/bin/python3
```

So having `brew` as part of the path for all sessions and loaded early is preferable.

Ideally we only add the new `/opt/homebrew` location if the machine is actually apple Silicon, but that's a trivial thing since if it's an intel-based box, that folder is unlikely to exist.

Given the above, it may be simplest to go with the solution in https://github.com/sorin-ionescu/prezto/pull/1974/files and be done with it.

Thoughts from others?

**Comment by belak** [comment]
We already have some path modifications in our provided zprofile, so I'd lean towards there... but I realize that might not work great for your use case.

Would the hack we include in .zshenv to force the .zprofile to be loaded in the case of non-interactive, non-login sessions be enough of a fix for you (on top of adding the path to .zprofile)?

**Comment by indrajitr** [comment]
I agree with @belak. `.zprofile` is probably the place. And when `.zprofile` (or even `.zshenv`) are loaded we don't assume anything about the environment and thus we don't have too many conditions in there (yes, `$BROWSER` is an exception :)).

So instead of scattering `$path` initialization in multiple places (`.zprofile` and `.zshenv`) OOTB, should we just keep the same convention (and just make sure the global `$path` isn't bloated unecessarily)?

#1997 has a version for review.

**Comment by jeffwidman** [comment]
One other reason **not** to put anything touching `$PATH` into `.zshenv`: 
* https://github.com/orgs/Homebrew/discussions/4747
* https://gist.github.com/jeffwidman/47dfc34c0ed504e27aa52beeaa5a981a
